pkg/controllers/scalablenodegroup/v1alpha1/suite_test.go

```diff
@@ -53,24 +54,76 @@ var _ = AfterSuite(func() {
 
 var _ = Describe("Examples", func() {
 	var ns *environment.Namespace
+	var defaultReplicaCount = ptr.Int32(3)
+
 	var sng *v1alpha1.ScalableNodeGroup
 
 	BeforeEach(func() {
 		var err error
 		ns, err = env.NewNamespace()
 		Expect(err).NotTo(HaveOccurred())
 		sng = &v1alpha1.ScalableNodeGroup{}
+		fakeCloudProvider.NodeGroupStable = true
+		fakeCloudProvider.NodeGroupMessage = ""
+		fakeCloudProvider.WantErr = nil
+		sng.Spec.Replicas = defaultReplicaCount
+		fakeCloudProvider.NodeReplicas[sng.Spec.ID] = ptr.Int32(0)
 	})
 
 	Context("ScalableNodeGroup", func() {
 		It("should be created", func() {
 			Expect(ns.ParseResources("docs/examples/reserved-capacity-utilization.yaml", sng)).To(Succeed())
-			sng.Spec.Replicas = ptr.Int32(5)
-
+			fakeCloudProvider.NodeReplicas[sng.Spec.ID] = ptr.Int32(*sng.Spec.Replicas)
 			ExpectCreated(ns.Client, sng)
 			ExpectEventuallyHappy(ns.Client, sng)
-
 			ExpectDeleted(ns.Client, sng)
 		})
 	})
+
+	Context("ScalableNodeGroup Reconcile tests", func() {
+		It("Test reconciler to scale up nodes", func() {
+			Expect(fakeController.Reconcile(sng)).To(Succeed())
+			Expect(fakeCloudProvider.NodeReplicas[sng.Spec.ID]).To(Equal(defaultReplicaCount))
+		})
+
+		It("Test reconciler to scale down nodes", func() {
+			fakeCloudProvider.NodeReplicas[sng.Spec.ID] = ptr.Int32(10) // set existing replicas higher than desired
+			Expect(fakeController.Reconcile(sng)).To(Succeed())
+			Expect(fakeCloudProvider.NodeReplicas[sng.Spec.ID]).To(Equal(defaultReplicaCount))
```

Comment:
You should compare data rather than pointers here. This might be easier if defaultReplicaCount was a value instead of a pointer.

Reply:
Yeah makes sense, I change it. Thanks.